lib/Doctrine/DBAL/Migrations/Configuration/Configuration.php

```diff
@@ -601,15 +601,15 @@ public function getRelativeVersion($version, $delta)
             $this->registerMigrationsFromDirectory($this->getMigrationsDirectory());
         }
 
-        $versions = array_keys($this->migrations);
-        array_unshift($versions, 0);
-        $offset = array_search($version, $versions);
+        $versions = array_map('strval', array_keys($this->migrations));
+        array_unshift($versions, '0');
+        $offset = array_search((string)$version, $versions);
         if ($offset === false || !isset($versions[$offset + $delta])) {
             // Unknown version or delta out of bounds.
             return null;
         }
 
-        return (string) $versions[$offset + $delta];
+        return $versions[$offset + $delta];
```

Comment:
What for? All values are already converted by `strval` and non-string case is covered by tests.
My idea was to fix type discrepancy as soon as possible and do not rely on extra casts.